src/wowman/core.clj

```diff
@@ -479,74 +536,51 @@
   (when-let [uri (:group-id addon-summary)]
     (-> uri java.net.URI. .getHost (clojure.string/split #"\.") second)))
 
-;; todo: this belongs in a database. whatever laziness I'm trying to do isn't working
-(defn -match-installed-addons-with-catalog
-  "for each installed addon, search the catalog across multiple joins until a match is found."
+
+;;
+
+
+(defn find-in-db [installed-addon toc-keys catalog-keys]
+  (let [catalog-keys (if (vector? catalog-keys) catalog-keys [catalog-keys]) ;; ["source" "source_id"] => ["source" "source_id"], "name" => ["name"]
+        sql-arg-template (clojure.string/join " AND " (mapv #(format "%s = ?" %) catalog-keys)) ;; "source = ? AND source_id = ?", "name = ?"
+
+        toc-keys (if (vector? toc-keys) toc-keys [toc-keys])
+        sql-arg-vals (mapv #(get installed-addon %) toc-keys) ;; [:source :source-id] => ["curseforge" 12345], [:name] => ["foo"]
+
+        sql (str select-*-catalog "where " sql-arg-template)
+        ;;_ (info sql-arg-vals) ;; there are cases where an arg is nil. should we still query if we don't have all the facts?
```

Comment:
this is still a valid question, address it

Reply:
I actually can't replicate this anymore. It might have been a problem early on with the database wrangling. Added a warning if it crops up again